src/main/java/org/openstreetmap/atlas/geography/boundary/CountryBoundaryMap.java

```diff
@@ -1118,6 +1121,55 @@ public void initializeGridIndex(final Set<String> countries)
         }
     }
 
+    /**
+     * Filters the given iterable of {@link ComplexBoundary}'s to only allow a single country to
+     * claim an outer boundary polygon.
+     *
+     * @param complexBoundaries
+     *            The {@link ComplexBoundary}'s to filter
+     * @return An iterable of {@link ComplexBoundary} with no duplicate outer polygons
+     */
+    public Iterable<ComplexBoundary> resolveOverlappingBorders(
+            final Iterable<ComplexBoundary> complexBoundaries)
+    {
+        final List<ComplexBoundary> dedupedComplexBoundaries = new ArrayList<>();
+        final Set<org.openstreetmap.atlas.geography.Polygon> processedOuters = new HashSet<>();
+        final List<ComplexBoundary> complexBoundariesList = Lists.newArrayList(complexBoundaries);
+        Collections.sort(complexBoundariesList, (firstBoundary, secondBoundary) ->
```

Comment:
the sorted collection here doesn't seem used for by the deduplication-- does it have a purpose that I'm not seeing? or should it be removed?

Reply:
Good catch! Meant to loop over the resulting `complexBoundariesList`. Fixing now.